classes/institution/Institution.inc.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * @defgroup institution Institution
+ * Implements institutions that are used for subscriptions and statistics.
+ */
+
+ /**
+ * @file classes/institution/Institution.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class Institution
+ * @ingroup institution
+ *
+ * @see DAO
+ *
+ * @brief Basic class describing an institution.
+ */
+
+namespace PKP\institution;
+
+class Institution extends \PKP\core\DataObject
+{
+    public const IP_RANGE_RANGE = '-';
+    public const IP_RANGE_WILDCARD = '*';
+
+    /**
+     * Get the context ID
+     */
+    public function getContextId(): int
+    {
+        return $this->getData('contextId');
+    }
+
+    /**
+     * Set the context ID
+     */
+    public function setContextId(int $contextId): void
+    {
+        $this->setData('contextId', $contextId);
+    }
+
+    /**
+     * Get the ROR
+     */
+    public function getROR(): ?string
+    {
+        return $this->getData('ror');
+    }
+
+    /**
+     * Set the ROR
+     */
+    public function setROR(string $ror): void
+    {
+        $this->setData('ror', $ror);
+    }
+
+    /**
+     * Get the localized name of the institution
+     */
+    public function getLocalizedName(string $preferredLocale = null): string
+    {
+        return $this->getLocalizedData('name', $preferredLocale);
+    }
+
+    /**
+     * Get the name of the institution
+     */
+    public function getName(string $locale = null): string|array
```

Comment:
Should be `?string` here and elsewhere, I think?

Reply:
It is not needed, because we have the default value and the user would use the function without parameters: `->getName()`
If we would like to use it like this `->getName(null)` and without the default value `public function getName(?string $locale) `then we would need it. Do we have a proffered way for these cases?

Reply:
I think this is a quirk of type hinting that I've looked up before. No change needed!